ddtrace/contrib/molten/patch.py

```diff
@@ -0,0 +1,143 @@
+
+        with pin.tracer.trace(func_name(wrapped), service=pin.service, resource=resource):
+            return wrapped(*args, **kwargs)
+
+    return _trace_func
+
+def trace_middleware(middleware):
+    @wrapt.function_wrapper
+    def _trace_middleware(wrapped, instance, args, kwargs):
+        pin = Pin.get_from(molten)
+        if not pin or not pin.enabled():
+            return wrapped(*args, **kwargs)
+
+        name = None
+        if inspect.isfunction(wrapped):
+            name = wrapped.__name__
+        else:
+            name = type(wrapped).__name__
+
+        with pin.tracer.trace(name, service=pin.service):
+            return trace_func(name)(wrapped(*args, **kwargs))
+
+    return _trace_middleware(middleware)
+
+def patch_start_response(start_response):
+    @wrapt.function_wrapper
+    def _start_response(wrapped, instance, args, kwargs):
+        pin = Pin.get_from(molten)
+        span = pin.tracer.current_root_span()
+        status, headers, exc_info = args
+        status_code, _, _ = status.partition(' ')
+        span.set_tag(http.STATUS_CODE, status_code)
+        return wrapped(*args, **kwargs)
+
+    return _start_response(start_response)
+
+def patch_add_route(wrapped, instance, args, kwargs):
+    def _wrap(route_like, prefix="", namespace=None):
+        # avoid patching non-Route, e.g. Include
+        if not isinstance(route_like, molten.Route):
+            return wrapped(*args, **kwargs)
+
+        resource = '{} {}'.format(
+            route_like.method,
+            route_like.name or route_like.template
+        )
+
+        # patch handler for route
+        route_like.handler = trace_func(resource)(route_like.handler)
+
+        return wrapped(route_like, prefix=prefix, namespace=namespace)
+
+    return _wrap(*args, **kwargs)
+
+def patch_app_call(wrapped, instance, args, kwargs):
+    pin = Pin.get_from(molten)
+    if not pin or not pin.enabled():
+        return wrapped(*args, **kwargs)
+
+    params, start_response = args
+    start_response = patch_start_response(start_response)
+    method = params.get('REQUEST_METHOD')
+    path = params.get('PATH_INFO')
+    resource = u'{} {}'.format(method, path)
+    with pin.tracer.trace(func_name(wrapped), service=pin.service, resource=resource) as span:
+        span.set_tag(http.METHOD, method)
+        span.set_tag(http.URL, path)
+        return wrapped(params, start_response, **kwargs)
+
+
+def patch_app_init(wrapped, instance, args, kwargs):
+    # allow instance to be initialized with middleware
+    wrapped(*args, **kwargs)
+
+    # add Pin to instance
+    pin = Pin.get_from(molten)
+
+    if not pin or not pin.enabled():
+        return
+
+    # trace middleware in instance
+    instance.middleware = [
+        trace_middleware(mw)
+        for mw in instance.middleware
+    ]
+
+    # patch class methods of component instances
+    for component in instance.components:
+        component.__class__.can_handle_parameter = \
```

Comment:
I would prefer this as:

```python
component.__class__.can_handle_parameter = trace_func(component.__class__.__name__)(
    component.can_handle_parameter
)
```

or

```python
wrapper = trace_func(resource)
component.__class__.can_handle_parameter = wrapper(component.can_handle_parameter)
```


There are other ways to do it too, I just want to avoid using `\` to break up lines.